File: Cloth-Store-Admin-FE/src/app/core/apis/Dtos/Product.model.ts
import {CategoryModel} from "./Category.model";
import {CompanyModel} from "./Company.model";
import {ColorsModel} from "./Colors.model";
import {SizesModel} from "./Sizes.model";

export class ProductModel {
  id!: string | null;
  code!: string | null;
  price!: number | null;
  name!: string | null;
  description!: string | null;
  status!: string | null;
  category!: CategoryModel | null;
  company!: CompanyModel | null;
  image!: string | null;
  specification!: string | null;
  colors!: ColorsModel[] | null;
  sizes!: SizesModel[] | null;
}

File: Cloth-Store-Agency-BE/ct-core-model/src/main/java/com/example/ctcoremodel/ProductModel.java
package com.example.ctcoremodel;

import com.example.ctcommon.enums.SpecificationProduct;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@Data
@NoArgsConstructor
public class ProductModel {
    private String id;
    private String code;
    private String name;
    private Double price;
    private String status;
    private String description;
    private String image;
    @Enumerated(EnumType.STRING)
    private SpecificationProduct specification;
    private CateloryModel catelory;
    private CompanyModel company;
}
